### PR TITLE
SONAR-7187 WS api/ce/activity handles queue and past CE tasks

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/ws/ActivityAction.java [hunk 1]
```diff
@@ -17,144 +17,162 @@
...
       .setSince("5.2");
-    action.createParam(PARAM_COMPONENT_UUID)
-      .setDescription("Optional id of the component (project) to filter on")
+
+    action.createParam(PARAM_COMPONENT_ID)
+      .setDescription("Id of the component (project) to filter on")
       .setExampleValue(Uuids.UUID_EXAMPLE_03);
     action.createParam(PARAM_COMPONENT_QUERY)
-      .setDescription(format("Optional search by component name or key. Must not be set together with %s.", PARAM_COMPONENT_UUID))
+      // TODO add like on name and exact match on key
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/ws/ActivityAction.java]: INFO Complete the task associated to this TODO comment. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1135)

File: server/sonar-server/src/main/java/org/sonar/server/computation/ws/ActivityAction.java [hunk 2]
```diff
@@ -177,18 +215,61 @@ private void checkPermissions(CeActivityQuery query) {
...
+      .setPageSize(request.mandatoryParamAsInt(Param.PAGE_SIZE));
+
+    checkRequest(activityWsRequest.getComponentId() == null || activityWsRequest.getComponentQuery() == null, "%s and %s must not be set at the same time",
+      PARAM_COMPONENT_ID, PARAM_COMPONENT_QUERY);
+
+    return activityWsRequest;
+  }
+
+  private static class Tasks {
+    private final Iterable<WsCe.Task> tasks;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/ws/ActivityAction.java]: MAJOR Rename field "tasks" [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1700)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 3 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'ce' package (org.sonarsource.sonarqube:sonar-ws:src/main/java/org/sonarqube/ws/client/ce) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'ce' package (org.sonarsource.sonarqube:sonar-ws:src/main/java/org/sonarqube/ws/client/ce) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: sonar-db/src/main/resources/org/sonar/db/ce/CeActivityMapper.xml [hunk 3]
```diff
@@ -67,8 +67,11 @@
           #{cUuid}
         </foreach>
       </if>
-      <if test="query.status != null">
-        and ca.status=#{query.status}
+      <if test="query.statuses != null">
```
Comment (simonbrandhof) [inline sonar-db/src/main/resources/org/sonar/db/ce/CeActivityMapper.xml]: be careful with H2, empty IN is not supported. I think you should test != null and !empty()

**Comment by simonbrandhof** [comment]
LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request closed without merging**